### PR TITLE
Develop covid

### PR DESCRIPTION
**Pull request by tnagorra**

### CODE REVIEW

File: MapContainer/index.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
-import React, { useContext } from 'react';
+import React, { useContext, useEffect } from 'react';
 import { _cs } from '@togglecorp/fujs';
 
 import { MapChildContext } from '../context';
-import styles from './styles.scss';
+import useDimension from '../useDimension';
+
+import styles from './styles.css';
+// import styles from './styles.scss';
```
Comment (tnagorra) [inline MapContainer/index.tsx:8]: No need to add un-necessary line

File: MapImage.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        imageOptions,
+        onLoad,
+    } = props;
+
+    const [initialName] = useState(name);
+    const [initialUrl] = useState(url);
+    const [initialImage] = useState(image);
+    const [initialImageOptions] = useState(imageOptions);
+
+    // Handle change in bounds
```
Comment (tnagorra) [inline MapImage.tsx:40]: How does this handle change in bounds?

File: MapShapeEditor.tsx [hunk 3]
```diff
@@ -16,22 +18,34 @@ interface ModeChangeEvent {
...
 }
 
+
 interface Props {
     geoJsons: mapboxgl.MapboxGeoJSONFeature[];
 
-    onCreate?: (geojsons: mapboxgl.MapboxGeoJSONFeature[]) => void;
-    onDelete?: (geojsons: mapboxgl.MapboxGeoJSONFeature[]) => void;
-    onUpdate?: (geojsons: mapboxgl.MapboxGeoJSONFeature[]) => void;
+    onCreate?: (geojsons: mapboxgl.MapboxGeoJSONFeature[], draw: Draw | undefined) => void;
```
Comment (tnagorra) [inline MapShapeEditor.tsx]: Should we not make sure Draw is defined?

File: MapSource/index.tsx [hunk 4]
```diff
@@ -60,8 +65,14 @@ const MapSource = (props: Props) => {
                 ? { ...initialSourceOptions, data: initialGeoJson }
                 : initialSourceOptions;
 
-            console.warn(`Creating new source: ${sourceKey}`);
-            map.addSource(sourceKey, options);
+            if (initialDebug) {
+                console.warn(`Creating new source: ${sourceKey}`, options);
+            }
+            try {
```
Comment (tnagorra) [inline MapSource/index.tsx:71]: Why are we catching an error?

File: MapSource/index.tsx [hunk 5]
```diff
@@ -85,25 +96,121 @@ const MapSource = (props: Props) => {
...
+            return () => {
+                // map.off('data', handleData);
+                map.on('move', updateMarkers);
+                map.on('moveend', updateMarkers);
+            };
+        },
+        [map, sourceKey, mapStyle, createMarkerElement, updateMarkers, geoJson],
+    );
+
+    useEffect(
```
Comment (tnagorra) [inline MapSource/index.tsx:201]: This is a hack. Need to document why we did it.

**Review by tnagorra — COMMENTED** [review]
Need to document the decisions.

File: MapSource/index.tsx (hunk 4 above)
Comment (frozenhelium) [reply MapSource/index.tsx:71]: Map no work if no catch

**Review by frozenhelium — APPROVED** [review]

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**